city.lua

```diff
@@ -846,7 +847,7 @@ local function startConstruction(city, buildingConstruction, queueIndex, allowed
             coordinates = Queue.popleft(city[queueIndex])
             if coordinates == nil then
                 -- If there are no more entries left in the queue, then abort
-                return false
+                goto failure
```

Comment:
i'll rework this `goto` later, too. it just was so simple to log failure reason here by something alike `try-except-finally`

@winex: move logging into sub-function`(..., reason)` and call before `return`